lib/shopify_cli/theme/app_extension.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+require_relative "file"
+require_relative "theme_admin_api"
+
+require "pathname"
+require "time"
+
+module ShopifyCLI
+  module Theme
+    class AppExtension
```

Comment:
@karreiro this follows the same method structure of Theme but I felt it didn't warrant a direct inheritance